app/core/component/forms/edit-course-form/edit-course-form.controller.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+angular
+    .module('editCourseForm')
+    .controller('EditCourseFormController', function($rootScope, courseService, eventsFactory) {
+        var $ctrl = this;
+        $ctrl.showFormForEdit = false;
+
+        $rootScope.$on(eventsFactory.courseExchangeWithEditForm, function(event, course) {
+            $ctrl.toEditCourse = course;
+            $ctrl.showFormForEdit = !$ctrl.showFormForEdit;
```

Comment:
I can change logic, as you say, in $ctrl.editCourse function inside this controller but in code above i can't do it because  $ctrl.showFormForEdit = !$ctrl.showFormForEdit; triggered form visibility on edit button click inside course-list component and user* can close form if he needs it.

Reply:
fixed